### ISSUE TITLE
🐛 BUG: D1 does not VACUUM causing performance issues

### ISSUE THREAD

**Issue by matthewjosephtaylor** [cloudflare/workerd#1618]
### Which Cloudflare product(s) does this pertain to?

D1

### What version(s) of the tool(s) are you using?

3.25.0

### What version of Node are you using?

20.9.0

### What operating system and version are you using?

Mac Darwin Kernel Version 23.0.0

### Observed behavior
D1 gets slower and slower as I use as I do a lot of inserts and deletes. the sqlite DB file inside .wrangler seems to grow endlessly

### Expected behavior
sqlite auto vacuum should probably be enabled.

### Steps to reproduce
Insert a lot of data into D1 db
delete a lot of rows from D1 db
Note that the size of the sqlite file on disk does not reduce.

Work-around:

```bash
sqlite3 .wrangler/state/v3/d1/<dbfile>.sqlite
# once opened
VACUUM;
.quit

```

**Comment by a-robinson** [comment]
We should be enabling auto-vacuum in local mode. cc @kentonv just in case he disagrees since he handled this for non-local D1.

**Comment by kentonv** [comment]
We enable auto-vacuum in production, we should probably just do the same in local workerd.

That said, while auto-vacuum will allow the database to shrink when rows are deleted, I am not sure about the "gets slower and slower" part. What is making things slower, exactly? I wouldn't expect excessive freelist size to affect performance. Is fragmentation causing the slowness? If so auto-vacuum could actually make it worse.

**Comment by thipperz** [comment]
Hey, I found this issue while searching for how to vacuum D1.

I just removed ~5M rows from my D1 db. What should I do? Is auto vacuum enabled in production already? Thanks!

**Comment by kentonv** [comment]
Yes, auto-vacuum has been enabled in production all along. You don't have to do anything.

It's only in local testing mode that it wasn't enabled. I can't remember if we fixed that or not -- but that only applies to local testing.
